From 4f57903bb93330e92c50776212699cca41985b2f Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 9 Feb 2024 11:46:40 +0100
Subject: [PATCH 0535/1695] plat-stm32mp1: stm32mp1_pmic: add LPLV_STOP2 mode

Add LPLV_STOP2 mode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git cdf322532b26f8bcc2af174e4b24c4225194f492 plat-stm32mp1: stm32mp1_pmic: add LPLV_STOP2 mode]
Change-Id: I3ab52a4d0b5435c4aa09f64b216ceaddc22998d1
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358730
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
index 42735b898..3ee84deee 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
@@ -127,6 +127,7 @@ enum regu_lp_state_id {
 	REGU_LP_STATE_STANDBY,
 	REGU_LP_STATE_MEM,
 	REGU_LP_STATE_MEM_LOWVOLTAGE,
+	REGU_LP_STATE_MEM_LOWVOLTAGE_CPUOFF,
 	REGU_LP_STATE_COUNT
 };
 
@@ -135,6 +136,7 @@ static struct regu_lp_state regu_lp_state[REGU_LP_STATE_COUNT] = {
 	[REGU_LP_STATE_STANDBY] = { .name = "standby-ddr-sr", },
 	[REGU_LP_STATE_MEM] = { .name = "lp-stop", },
 	[REGU_LP_STATE_MEM_LOWVOLTAGE] = { .name = "lplv-stop", },
+	[REGU_LP_STATE_MEM_LOWVOLTAGE_CPUOFF] = { .name = "lplv-stop2", },
 };
 
 static unsigned int regu_lp_state2idx(const char *name)
-- 
2.47.2

